lib/src/css_mirroring/rulesets_processor.dart

```diff
@@ -0,0 +1,82 @@
+part of scissors.src.css_mirroring.bidi_css_generator;
+
+abstract class NodeProcessor {
+  void process(EditConfiguration e, TopLevelsPair t, int i);
+}
+
+class DeclarationProcessor extends NodeProcessor {
+  final _PendingRemovals removableRuleSets;
+  final int parentEnd;
+
+  DeclarationProcessor(this.removableRuleSets, this.parentEnd);
+
+  @override
+  void process(EditConfiguration _editConfig, TopLevelsPair _topLevelsPair,
+      int _iTopLevel) {
+    final removals = new _PendingRemovals(_editConfig.transaction);
```

Comment:
If this were a function that takes a transaction (+ edit config, where the transaction has nothing to do), would be simpler here IMO